apis/hive/v1/hiveconfig_types.go

```diff
@@ -41,6 +41,14 @@ type HiveConfigSpec struct {
 	// +optional
 	GlobalPullSecretRef *corev1.LocalObjectReference `json:"globalPullSecretRef,omitempty"`
 
+	// HiveImagePullSecretRef is used to specify a pull secret that can be used to pull Hive's own image.
+	// If hive has been deployed from a private registry, cluster installations will not succeed unless
+	// this reference is specified. This secret must live in Hive's TargetNamespace.
+	// NOTE: This secret will be copied into the namespace of every ClusterDeployment, overwriting any secret
+	// with the same name.
```

Comment:
I don't think this is true either, as it looks like you're naming those Secrets after the cd/deprovision.

Though thinking about it, I think I like this idea better. There's nothing stopping multiple CDs being created in the same namespace, and we would rather have just one copy of this pull secret per namespace.

Reply:
I made it per-CD because then k8s will GC the secret when the CD no longer exists.  If we're okay with leaving the dangling secret in a CD namespace, then we can get away with one per NS

Reply:
Per discussion: Let's try OwnerReferences per CD for the (single) Secret and see if that works.